src/main/java/dk/sdu/se_f22/sharedlibrary/models/ProductHit.java

```diff
@@ -0,0 +1,162 @@
+        this.uuid = UUID.fromString(stringId);
+
+        this.averageUserReview = Double.parseDouble(product.get(ProductAttribute.AVERAGE_USER_REVIEW));
+        this.inStock = product.getLocations();
+        this.ean = Long.parseLong(product.get(ProductAttribute.EAN));
+        this.price = Double.parseDouble(product.get(ProductAttribute.PRICE));
+        this.publishedDate = Instant.parse(product.get(ProductAttribute.PUBLISHED_DATE) + 'Z');
+        this.expirationDate = Instant.parse(product.get(ProductAttribute.EXPIRATION_DATE) + 'Z');
+        this.category = product.get(ProductAttribute.CATEGORY);
+        this.name = product.get(ProductAttribute.NAME);
+        this.description = product.get(ProductAttribute.DESCRIPTION);
+
+        if (!product.get(ProductAttribute.SIZE).equals("unavailable")){
+            this.size = product.get(ProductAttribute.SIZE);
+        }
+
+        if (!product.get(ProductAttribute.CLOCKSPEED).equals("unavailable")){
+            this.clockspeed = Double.parseDouble(product.get(ProductAttribute.CLOCKSPEED));
+        }
+
+        if (!product.get(ProductAttribute.WEIGHT).equals("unavailable")){
+            this.weight = Double.parseDouble(product.get(ProductAttribute.WEIGHT));
+        }
+    }
+
+    /**@param attributeName The name of the attribute to retrieve
+     * @return The value of the attribute referenced
+     * @throws IllegalArgumentException if the attribute input does not correspond to any of the attributes, whose value is a double
+     */
+    public double getDoubleValue(String attributeName){
+        return switch (attributeName) {
+            case ("price") -> this.getPrice();
+            case ("averageUserReview") -> this.getAverageUserReview();
+            case ("clockspeed") -> this.getClockspeed();
+            case ("weight") -> this.getWeight();
+            default -> throw new IllegalArgumentException(attributeName + "does not exist as a double attribute: ");
+        };
+    }
+
+    public UUID getUuid() {
+        return uuid;
+    }
+
+    public double getAverageUserReview() {
+        return averageUserReview;
+    }
+
+    public List<String> getInStock() {
+        return inStock;
+    }
+
+    public long getEan() {
+        return ean;
+    }
+
+    public double getPrice() {
+        return price;
+    }
+
+    public Instant getPublishedDate() {
+        return publishedDate;
+    }
+
+    public Instant getExpirationDate() {
+        return expirationDate;
+    }
+
+    public String getCategory() {
+        return category;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public String getSize() {
+        return size;
+    }
+
+    public double getClockspeed() {
+        return clockspeed;
+    }
+
+    public double getWeight() {
+        return weight;
+    }
+
+    public long getLongValue(String productAttribute) {
```

Comment:
What is the use of this method?
You have a getEan method? So i'm not sure I understand the purpose of it.

Reply:
they will be used for accesing the attributes when we read their names from the database i.e.
String productAttribute = "ean"
or 
productAttribute = "weight"

This allows us the flexibility of not being bound to only the attributes currently present in products, but expansion without having to rewrite the code.
The goal is that the customer can add productAttribute without requiring us to add new Filter classes to handle this.

I see how it is currently unclear, since the pseudocode, to show this use has somehow vanished.
I will attempt to find it and add it to clarify (as comments of course)